ark/spLDA/processing_test.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import pytest
+
+import ark.spLDA.processing as pros
+import ark.settings as settings
+from ark.utils.misc_utils import verify_in_list
+from ark.utils.test_utils import make_cell_table
+
+# Generate, format, and featurize a test cell table
+N_CELLS = 1000
+TRAIN_FRAC = 0.75
+TRAIN_CELLS = TRAIN_FRAC * N_CELLS
+TEST_CELL_TABLE = make_cell_table(N_CELLS)
+TEST_FORMAT = pros.format_cell_table(
+    cell_table=TEST_CELL_TABLE, clusters=list(np.unique(TEST_CELL_TABLE[settings.CLUSTER_ID])))
+TEST_FEATURES = pros.featurize_cell_table(cell_table=TEST_FORMAT, train_frac=TRAIN_FRAC)
+
+
+def test_format_cell_table():
+    # Check that number of FOVS match
+    fov_list=[x for x in TEST_FORMAT.keys() if x not in ['fovs', 'markers', 'clusters']]
+    verify_in_list(
+        fovs1=list(np.unique(TEST_CELL_TABLE[settings.FOV_ID])), fovs2=fov_list)
+    # Check that columns were retained/renamed
+    verify_in_list(
+        cols1=["x", "y", "cluster_id", "cluster", "is_index"], cols2=list(TEST_FORMAT[1].columns))
+    # Check that columns were dropped
+    assert len(TEST_CELL_TABLE.columns) > len(TEST_FORMAT[1].columns)
+
+
+def test_featurize_cell_table():
+    # Check for consistent dimensions
+    assert TEST_FEATURES["featurized_fovs"].shape[0] == TEST_CELL_TABLE.shape[0]
+    assert TEST_FEATURES["featurized_fovs"].shape[0] == N_CELLS
+    assert TEST_FEATURES["train_features"].shape[0] == TRAIN_CELLS
+
+
+def test_compute_topic_eda():
+    # at least 25 bootstrap iterations
+    with pytest.raises(ValueError, match="Number of bootstrap samples must be at least"):
+        pros.compute_topic_eda(TEST_FEATURES["featurized_fovs"], topics=[5], num_boots=20)
+    # appropriate range of topics
+    with pytest.raises(ValueError, match="Number of topics must be greater than 2"):
+        pros.compute_topic_eda(TEST_FEATURES["featurized_fovs"], topics=[2], num_boots=25)
+    with pytest.raises(ValueError, match=r"Number of topics must be less than"):
+        pros.compute_topic_eda(TEST_FEATURES["featurized_fovs"], topics=[1000], num_boots=25)
```

Comment:
These are great. Should also add a test that the output contains the correct items. This will ensure that the function runs from start to end without errors